torch/csrc/autograd/variable.cpp

```diff
@@ -181,19 +181,29 @@ void rebase_history(const Variable& self, Edge gradient_edge) {
   }
 
   set_gradient_edge(self, std::move(gradient_edge));
+  if (self.retains_grad()) {
```

Comment:
There can be early return in this functions so this is not enough.

Reply:
Yup, will indeed need to handle the in-place over view case